### DISCUSSION TITLE
Where does print go on a worker?

### DISCUSSION THREAD

**Discussion by jsignell** [dask/dask#7202] [Q&A]
I have been staring at this for a while and can't figure it out. When I call `print` in a task within a distributed worker, where does that print happen?

When I use a `LocalCluster` from Jupyter it prints in the terminal where I started my Jupyter from. If I want it to instead print to my logs I can do:

```python
from dask.distributed import Client
from distributed.worker import logger

client = Client()

def log_it():    
    logger.info("This will output to the logs!")
    
client.run(log_it)
```

What I'd like is for this to print in my worker logs as well.

```python
from dask.distributed import Client

client = Client()

def print_it():    
    print("Print this in the logs!")
    
client.run(print_it)
```

**Comment by quasiben** [comment]
When you print it should go to stdout of the worker.  Those won't typically be reflected in the logs captured by dask for example at say: `http://localhost:8787/info/logs/tcp%3A%2F%2F127.0.0.1%3A53082.html`.  I think what you have down is correct but maybe there is a config setting in place for the worker ?  Could you try with info/debug/error/warn ?

xref: https://docs.dask.org/en/latest/debugging.html#logs for configuring client/scheduler/worker logs

**Reply by jrbourbeau** [reply]
FWIW there was a discussion about capturing stdout/stderr on workers and redirecting to into Dask's logging system at some point (xref https://github.com/dask/distributed/issues/2033)

**Reply by jsignell** [reply]
Ah thanks yeah. I came across that issue in my search, but neglected to mention it. I had trouble telling if the information in that issue was uptodate.

**Reply by jsignell** [reply]
@quasiben I tried setting worker logging to debug and it is more verbose, but still doesn't capture the print output.

**Comment by jsignell** [comment]
I opened a PR to discuss this more: https://github.com/dask/distributed/pull/4502
